src/core/xc_fallback.jl

```diff
@@ -0,0 +1,74 @@
+# TODO Temporary, until Libxc.jl is used
+@enum FunctionalFamily begin
+    family_unknown  = -1
+    family_lda      = 1
+end
+mutable struct Functional
+    name::String
+    family::FunctionalFamily
+    n_spin::Int
+end
+function Functional(identifier)
+    Functional(string(identifier), family_lda, 1)
+end
+# TODO end temporary
+
+
+function lda_x!(ρ; α=1, E=nothing, Vρ=nothing)
+    tmp = -α * cbrt.(3/π * ρ)
+
+    if E !== nothing
+        E .= 3/4 .* tmp
+    end
+    if Vρ !== nothing
+        Vρ .= tmp
+    end
+end
```

Comment:
This complexifies the code without adding too much functionality, and kwargs are tricky for inference. I think I would write this as `lda_x!(E,V,rho;alpha)`, since computing E and computing V are about the same cost; and then maybe a helper function `lda_x(E,rho;alpha) = lda_x!(E,similar(rho),alpha)` for convenience?

Reply:
Huh, so actually I'm confused; is rho meant to be scalar or vector? If scalar, it should just return the value (eg return (E,V)), not modify anything.

Reply:
What then for higher derivatives, which will need to be computed only sometimes? In the [Libxc.jl](https://github.com/unkcpz/Libxc.jl) julia package, I solved this using kwargs, which (if present) indicate a computation, otherwise not. See the code: [evaluate.jl](https://github.com/unkcpz/Libxc.jl/blob/master/src/evaluate.jl).

`ρ` is meant to be a vector.

Reply:
I left that unchanged so far. Happy to discuss, however.

Reply:
I worry that this will be very slow. Also, using in-place is a bit inconvenient. But probably it doesn't really matter... One thing that might be useful is named tuples: `(a=1,b=2)`. Anyway, I don't have anything better to propose and it's pretty low-level anyway so let's go with that!

Reply:
Named tuple is not a bad idea ... on the other hand: I wonder why you fear the kwargs version is slow. From my tests it does not appear as type unstable and unused parts of the function (where the argument is `nothing`) are completely removed in the `@code_llvm` output as far as I can tell. So that sounds exactly as one would want it.

Reply:
OK! My fear appears outdated, that's great!

Reply:
Yeah, that's https://github.com/JuliaLang/julia/issues/9551, gone since feb 2018

Reply:
Great. Good to hear about that ... I guess then we are good to go for now.